components/resource/texturemanager.cpp

```diff
@@ -169,7 +169,17 @@ namespace Resource
             std::string ext;
             if (extPos != std::string::npos && extPos+1 < normalized.size())
                 ext = normalized.substr(extPos+1);
-            osgDB::ReaderWriter* reader = osgDB::Registry::instance()->getReaderWriterForExtension(ext);
+
+#ifdef USE_TGA
+            osgDB::ReaderWriter* reader =NULL;
+            if (ends_with(ext,"dds"))
+              reader = osgDB::Registry::instance()->getReaderWriterForExtension("tga");
+            else
+              reader = osgDB::Registry::instance()->getReaderWriterForExtension(ext);
```

Comment:
Consider
```c++
osgDB::ReaderWriter * const reader
(osgDB::Registry::instance()->getReaderWriterForExtension(ends_with(ext,"dds") ? "tga" : ext));
```
or similar, in both places in this file. Just looking to minimize duplication and keep things const.